src/ImageViewer/events.ts

```diff
@@ -0,0 +1,219 @@
+  "dataabort",
+  "sourcedataabort",
+  // this is fired on layers, not the map
+  // keeping it for reference
+  // "metadata",
+] as const;
+
+export const IMAGE_VIEWER_EVENT_TYPES = [
+  ...BASE_MAP_EVENT_TYPES,
+  ...ERROR_EVENTS,
+  ...RESIZE_EVENTS,
+  ...WEBGL_CONTEXT_EVENTS,
+  ...CAMERA_EVENTS,
+  ...UI_EVENTS,
+  ...DATA_EVENTS,
+  ...COOPERATIVE_GESTURE_EVENTS,
+];
+
+type BaseMapEventKeys = (typeof BASE_MAP_EVENT_TYPES)[number];
+type UiEventKeys = (typeof UI_EVENTS)[number];
+type CameraEventKeys = (typeof CAMERA_EVENTS)[number];
+type ErrorEventKeys = (typeof ERROR_EVENTS)[number];
+type ResizeEventKeys = (typeof RESIZE_EVENTS)[number];
+type WebglContextEventKeys = (typeof WEBGL_CONTEXT_EVENTS)[number];
+type DataEventKeys = (typeof DATA_EVENTS)[number];
+type CooperativeGestureEventKeys = (typeof COOPERATIVE_GESTURE_EVENTS)[number];
+
+export type ImageViewerEventTypes =
+  | BaseMapEventKeys
+  | UiEventKeys
+  | CameraEventKeys
+  | ErrorEventKeys
+  | ResizeEventKeys
+  | WebglContextEventKeys
+  | DataEventKeys
+  | CooperativeGestureEventKeys
+  | "imageviewerready"
+  | "imagevieweriniterror";
+
+// Properties to exclude when forwarding events
+const FORBIDDEN_EVENT_VALUES = ["lngLat", "_defaultPrevented"];
+
+type LngLatToPixel = (lngLat: LngLat) => [number, number];
+
+interface SetupGlobalMapEventForwarderOptions {
+  map: Map;
+  viewer: ImageViewer;
+  lngLatToPx: LngLatToPixel;
+}
+
+/**
+ * Sets up event forwarding from Map to ImageViewer with proper categorization
+ * @param {SetupGlobalMapEventForwarderOptions} options - The options for setting up the event forwarder.
+ * @param {Map} options.map - The map to forward events from.
+ * @param {ImageViewer} options.viewer - The viewer to forward events to.
+ * @param {LngLatToPixel} options.lngLatToPx - A function to convert LngLat to pixel coordinates.
+ * @returns {void}
+ */
+export function setupGlobalMapEventForwarder({ map, viewer, lngLatToPx }: SetupGlobalMapEventForwarderOptions): void {
+  IMAGE_VIEWER_EVENT_TYPES.forEach((eventType: Exclude<ImageViewerEventTypes, "imageviewerready" | "imagevieweriniterror">) => {
+    try {
+      map.on(eventType, (e: MapEventType[Exclude<ImageViewerEventTypes, "imageviewerready" | "imagevieweriniterror">]) => {
+        // Handle UI Events (mouse/touch interactions with coordinate transformation)
+        const uiEventName = eventType as UiEventKeys;
+        if (UI_EVENTS.includes(uiEventName)) {
+          const event = e as MapEventType[UiEventKeys];
+          const px = event.lngLat && lngLatToPx(event.lngLat);
+          const data = Object.entries(e).reduce(
+            (acc, [key, value]) => {
+              if (FORBIDDEN_EVENT_VALUES.includes(key)) {
+                return acc;
+              }
+              return {
+                ...acc,
+                [key]: value,
+              };
+            },
+            {
+              imageX: px[0],
+              imageY: px[1],
+            },
+          );
```

Comment:
I'd prefer `fromEntries` rather than `reduce`
```diff
-          const data = Object.entries(e).reduce(
-            (acc, [key, value]) => {
-              if (FORBIDDEN_EVENT_VALUES.includes(key)) {
-                return acc;
-              }
-              return {
-                ...acc,
-                [key]: value,
-              };
-            },
-            {
-              imageX: px[0],
-              imageY: px[1],
-            },
-          );
+          const data = {
+            imageX: px[0],
+            imageY: px[1],
+            ...Object.fromEntries(Object.entries(e).filter(([key]) => !FORBIDDEN_EVENT_VALUES.includes(key))),
+          };
```